### DISCUSSION TITLE
Made a tutorial for making Word by word Subtitle using the Subtitle Edit software

### DISCUSSION THREAD

**Discussion by sherkhan30452** [SubtitleEdit/subtitleedit#8066] [Show and tell]
I would like to thank you so much for making such an awesome software! 😀

[Create Word by word Subtitles for YT shorts | Video Editing Tutorial | ft. Subtitle Edit and Buzz](https://www.youtube.com/watch?v=U7H39cErhRo)
![Thumbnail 2](https://github.com/SubtitleEdit/subtitleedit/assets/33783004/f44bae2b-9eb4-4042-968a-9c40057bed70)

https://www.youtube.com/watch?v=U7H39cErhRo

**Comment by LeonCheung** [comment]
Cool. I was always trying to do it in SE with Whisper but failed. I tried to make it in Premiere, it can be done by setting parameters like this:

![pp](https://github.com/SubtitleEdit/subtitleedit/assets/494788/540882c3-ba34-4ab3-acdf-0c63765af2fa)

Not sure whether it could be done all in SE but thanks for the nice workaround with Buzz.

**Reply by Purfview** [reply]
You can do it with [Standalone Faster-Whisper](https://github.com/Purfview/whisper-standalone-win) too.

**Comment by niksedk** [comment]
It's possible in SE via this argument `--one_word 1` when using engine "Purfview's Faster Whisper":

![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/98bd0697-a47b-4d1d-8578-1b6e787af953)

Comment 1: In the ASSA style window, if you delete "Default" and only keep the "For shorts", then this style will be auto applied to all lines.

Comment 2: In the click on the "Layouts" icon in the toolbar, you can switch to a more shorts friendly view:
![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/0b74ffe5-2fd8-49c9-bae4-f5d1c7149609)

**Reply by sherkhan30452** [reply]
Actually I tried (in previous version), the argument you mentioned didn't work _(and not just once, multiple times, but failed)_ Since, it's wispher bug, I didn't thought I should report it as bug for subtitle edit software.

But I will surely try again, with new version of subtitle edit.

_Also thank you the layout tip_ .😃👍🏻

**Reply by Purfview** [reply]
> Actually I tried (in previous version), the argument you mentioned didn't work _(and not just once, multiple times, but failed)_ Since, it's wispher bug, I didn't thought I should report it as bug for subtitle edit software.

Not a bug, you need to disable SE's "post-processing". 
`--one_word 2` is recommended instead of `--one_word 1`

**Reply by sherkhan30452** [reply]
Ah, thank you for pointing it out, that's why it was not working.

Cool now it works, I need to make updated tutorial on this sooner.

**Comment by sherkhan30452** [comment]
@niksedk Hi, someone asked a question regarding the youtube video I posted 

> Create Word by word Subtitles for YT shorts | Video Editing Tutorial | ft. Subtitle Edit and Buzz

Comment:
`Hi. Feature question here. It's there a way to add animated subtitles? I mean, for example, that the words appear in their place on a sentence as they are spoken?`

Is it possible?

**Comment by niksedk** [comment]
You can add some ASSA animations for sure :)

You can try right-click in the list view, and choose "ASSA tools" - "Apply custom override tags":
![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/ea0ad5ad-721c-4e98-8eb9-9d3325be1a23)

In latest beta I've added a few templates:
![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/30b46544-75f5-4270-b4b0-31e53023464c)

https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.4/SubtitleEditBeta.zip

**Comment by niksedk** [comment]
@sherkhan30452: Perhaps a new video with SE 4.0.5 - whisper single word + applied ASSA effects

![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/b961d33e-3ff8-4ef7-bd19-912d8393047b)




https://github.com/SubtitleEdit/subtitleedit/assets/1261331/8c74248d-45ad-411e-8f25-9687a715542e

Or multiple effects combined:


https://github.com/SubtitleEdit/subtitleedit/assets/1261331/9d58497b-03aa-4f72-bf20-05d7b8697d00

**Reply by sherkhan30452** [reply]
Sure, I will make a tutorial as soon as I get time. Thanks for sharing the demo. And you mean 4.0.5 not 5.0.5, correct?

**Reply by niksedk** [reply]
Ah yes, 4.0.5 :)

**Comment by sherkhan30452** [comment]
@niksedk Just want to confirm, can it do like a Karaoke Video? Like in this video https://youtu.be/4YTIaMeKXts?t=450
I noticed the template from text color from white to red, however I don't to how to set word by word basis.

**Reply by sherkhan30452** [reply]
Reference - https://aegisub.org/docs/latest/ass_tags/
![image](https://github.com/SubtitleEdit/subtitleedit/assets/33783004/58b3949f-5db1-41fb-aa6f-eb37ed2c93f8)

**Comment by LeonCheung** [comment]
@sherkhan30452, it would not be hard to make that right now even without any further SE development. Try this:

1. Create a custom text format template like this:
![image](https://github.com/SubtitleEdit/subtitleedit/assets/494788/7876cc8a-21c1-4a6a-97b3-1ee64084f4ee)

BTW, @niksedk Not sure if this is a bug, but zz doesn't work as I want to export centiseconds as ASSA Karaoke needs that.

2. No need to hit Export then, simply select all texts in preview window. Then go back to the list view, select line1, then **right click** > **Column** > **Paste from clickboard**:
![image](https://github.com/SubtitleEdit/subtitleedit/assets/494788/ccc6f211-5dd3-4f48-bd61-842c1fdf6405)

3. Select lines that are going to be merged as a single line, then **right click > Merge selected lines**. Now you should have the ASSA Karaoke effect created.

NOTE: Before any improvements happen, you have to trim the duration values for \K tags, like this:

![image](https://github.com/SubtitleEdit/subtitleedit/assets/494788/19f9bd0a-6589-4f7a-9d0d-cc60955d76e6)

Then you may get what you want:
![image](https://github.com/SubtitleEdit/subtitleedit/assets/494788/f410cfdd-3841-4684-99ca-10261182cf0f)


---

However, @niksedk maybe a more fancy way could be from you with a similar concept?  😄  something like:

![image](https://github.com/SubtitleEdit/subtitleedit/assets/494788/6c5b449a-6eb5-4ceb-b924-f8ed342940bd)

**Oh guys, there could be unlimited ideas on mimicking popular effects that are widely used on platforms like TikTok. Millions of people post captioned videos everyday. It's a great chance to rock the world again with SE.**

**Comment by niksedk** [comment]
You can do karaoke stuff very fast with this setting (high-light current word):
![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/37bcb502-6a7c-4f48-ae7a-0bfce78d93d6)

I've replaced the under line with red:

https://github.com/SubtitleEdit/subtitleedit/assets/1261331/d6058c72-8eeb-4b3f-8021-e71515441620

**Reply by sherkhan30452** [reply]
I was able to change color, however how to change back, that I didn't figured it out, can you please share the syntax?

`I did something like this Replace "<u>" with "<color1>" and "</u>" with "<color2>". What did I do wrong?`

**Reply by sherkhan30452** [reply]
I figured it out, I was using color order opposite way.

**Reply by Josua-92** [reply]
How did you replace the highlighted word from an underline to a color?
Can you provide the steps you followed?
